api/app/services/jwt_service.rb

```diff
@@ -0,0 +1,34 @@
+class JwtService
+  ALGORITHM = "HS256"
+  ACCESS_TTL = 30.minutes
+  REFRESH_TTL = 7.days
+
+  class ExpiredToken < StandardError; end
+  class InvalidToken < StandardError; end
+
+  def self.secret_key
+    ENV.fetch("JWT_SECRET_KEY")
+  end
+
+  def self.encode(payload, exp:)
+    body = payload.dup
+    body[:exp] = exp.to_i
+    JWT.encode(body, secret_key, ALGORITHM)
+  end
+
+  def self.decode(token)
+    decoded, = JWT.decode(token, secret_key, true, { algorithm: ALGORITHM })
+    decoded.with_indifferent_access
+  rescue JWT::ExpiredSignature
+    raise ExpiredToken, "Token has expired"
+  rescue JWT::DecodeError
+    raise InvalidToken, "Token is invalid"
```

Comment:
The decode method does not handle all potential JWT errors, such as JWT::VerificationError. Rescue additional JWT errors to avoid unhandled exceptions.

```
rescue JWT::ExpiredSignature
  raise ExpiredToken, "Token has expired"
rescue JWT::DecodeError, JWT::VerificationError
  raise InvalidToken, "Token is invalid"
```